### ISSUE TITLE
Inconsistent behavior of `simplify` between versions of libgeos

### ISSUE THREAD

**Issue by kwikwag** [shapely/shapely#2134]
## Edit

After the interaction below, it seems the difference between installations is in the version of GEOS used (3.11 vs 3.12).

## Expected behavior and actual behavior.

I use `simplify` to approximate a quadrangle from my Polygon.
The algorithm tries to simplify the polygon iteratively until it either reaches a quadrangle or fails.

For some reason, on a large dataset of polygons, it does perfectly well on Windows, but fails rather consistently on Linux. The same version of Shapely is installed on both, as well as the same version of NumPy.
 
I expect both versions to work in the same manner (either both fail, or both succeed).

## Steps to reproduce the problem.

```python
import numpy as np
import shapely

data = [[3688.03, 2459.2], [3430.5, 2465.6], [3334.06, 2466.07], [3093.01, 2476.38], [2611.39, 2488.65], [2364.93, 2492.09], [2000.16, 2485.71], [1658.95, 2479.32], [1530.33, 2471.96], [1325.6, 2472.45], [949.54, 2464.6], [699.15, 2463.61], [694.74, 2460.18], [664.79, 659.88], [681.48, 656.44], [1628.52, 648.1], [1787.58, 646.13], [2151.37, 640.24], [2442.99, 638.28], [2653.12, 639.26], [3675.27, 661.35], [3666.92, 1157.21]]

def appx_quad(poly):
  poly = shapely.Polygon(poly)
  poly = poly.convex_hull
  appx = 1 / (2**10)
  while appx < 1 / 8 and len(poly.exterior.coords) >= 6:
    poly = poly.simplify(appx * poly.length, preserve_topology=False)
    appx *= 2
  return np.array(poly.exterior.coords[:-1])

print(appx_quad(data))
```

Output on Linux:

```
[[2442.99  638.28]
 [ 664.79  659.88]
 [ 694.74 2460.18]
 [3688.03 2459.2 ]
 [3675.27  661.35]]
```

Output on Windows:
```
[[ 664.79  659.88]
 [ 694.74 2460.18]
 [3688.03 2459.2 ]
 [3675.27  661.35]]
```

P.S. There should be quite a few examples more - as I said, this consistently fails in Linux but works wonderfully on Windows.

## Operating system

Windows 10 (conda) vs Amazon Linux 2 (pip) / Google Colab (pip)

## Shapely version and provenance

v2.0.1 on Windows 10 (conda) and Amazon Linux 2 (pip) [NumPy is 1.23.5]
v2.0.6 on Google Colab [NumPy is 1.26.4]

**Comment by kwikwag** [comment]
Apparently if I remove the call to `convex_hull` I get the same results in both operating systems.

**Comment by kwikwag** [comment]
However, this hugely simplified version without iterations and without the call to `convex_hull` still has the problem:

```python
import numpy as np
import shapely

data = [[2442.99,  638.28],
       [ 664.79,  659.88],
       [ 694.74, 2460.18],
       [2364.93, 2492.09],
       [3688.03, 2459.2 ],
       [3675.27,  661.35],
       [2442.99,  638.28]]

def appx_quad(poly, appx=1/16):
  poly = shapely.Polygon(poly)
  appx_poly = poly.simplify(appx * poly.length, preserve_topology=False)
  return np.array(appx_poly.exterior.coords[:-1])

print(appx_quad(data))
```

**Comment by theroggy** [comment]
Shapely is mainly a python wrapper around the GEOS library, so both the calculation of convex_hull as the simplification actually are done in by GEOS. Installing with conda versus pip will often lead to other versions of GEOS being installed. Please check the GEOS version with `shapely.geos_version_string`.

**Comment by kwikwag** [comment]
`shapely.geos_version_string`:
Amazon Linux 2 - '3.11.1'
Google Colab - `3.11.4`
Windows - '3.12.0'

I'll also add that once I remove `convex_hull`, most of the time the algorithm completes correctly. That is, even though the second script I sent doesn't use it (at least not directly anymore).

**Comment by kwikwag** [comment]
Indeed, when rebuilding Shapely on Google Colab with geos backport, I get the same result as Windows. So this seems to be a fix in GEOS in vesrion 3.12.0? Adding the commands for installing the backport PPA as well as reinstalling Shapely, for future reference:

```
!add-apt-repository ppa:savoury1/backports
!apt install libgeos3.12.0 geos-bin libgeos-dev
!pip uninstall -y shapely
!pip install -y shapely --no-cache-dir --no-binary shapely
```

**Comment by theroggy** [comment]
Recently released shapely 2.1 includes wheels with a recent GEOS version that solves this.
